viosock/sys/viosock.inx

```diff
@@ -56,6 +55,15 @@ WdfCoInstaller$KMDFCOINSTALLERVERSION$.dll = 1 ; make sure the number matches wi
 [VirtioSocket_Device.NT]
 CopyFiles=Drivers_Dir,VirtioSocket_Lib_CopyFiles
 
+ [VirtioSocket_Device.10.0.15063.Software]
+ AddSoftware=WSKWSP,,VirtioSocket_Software
```

Comment:
1. How do you expect to sign this driver for Windows Server 2016? After applying a Cumulative Update for Windows Server 2016 for x64-based Systems (KB4598243), the latest OS Build is 14393.4169.

2. According to https://learn.microsoft.com/en-us/windows-hardware/drivers/install/inf-addsoftware-directive: `Each AddSoftware directive describes the installation of standalone software. This directive should be used in an INF file of the **SoftwareComponent** setup class. `
How will this work in this INF with `Class=System`?

Reply:
It seems that the `SoftwareComponent` class is not a requirement. For example, INF for NVDA Display Driver on my machine uses the `AddSoftware` directive but is of class `Display`.

Reply:
1) well, I suppose there need to be special INF sections for handling OS versions that do not support the `AddSoftware` directive. I am currently experimenting with this.
Unfortunately, it is not possible to register the Winsock Provider by just adding certain keys and values into the Registry (like in `viorng` case); the `AddReg` directive is not strong enough for that.